### PR TITLE
Connected Register and Login Pages with Backend with direction after submission PRA-60 PRA-49

### PR DESCRIPTION
**Pull request by AnnaSolovykh**
PRA-49 

## Requirements

Send POST requests to backend to enable user to sign up and sign in successfully.

## Notes

Please follow password instructions carefully, as checking user's following them correctly with flash messages is our next coding challenge. Now you get status 500 in the console if you have chosen the password not due to the instructions. More info you can find in this case only in the backend terminal.

## Test Steps

When you open /register page, you can sign up successfully and get "status 201" in the console and see a new user in your Mongo DB.
After that you can open /login page, you can sign in successfully and get "status 200" in the console. 

## UI/UX Outcome

Screen Shot 2023-07-28 at 14 07 24
Screen Shot 2023-07-28 at 14 07 54

## Checklist

- The Linear ID is in the PR title
- My code follows best practices
- Documentation is included where needed

PRA-60
## Requirements

On successful submission, redirect to home page.
On unsuccessful submission, display an error message to the user.

## Notes

None

## Test Steps

After successful registration/login, you get redirected to the Home page. 
After unsuccessful registration/login, you get redirected to the Error page. 
From the Error page you can get either to the Login or Register page. 

## UI/UX Outcome
Screen Shot 2023-07-28 at 14 20 32
Screen Shot 2023-07-28 at 15 17 49

## Checklist

- The Linear ID is in the PR title
- My code follows best practices
- Documentation is included where needed

### CODE REVIEW

File: src/components/Login.tsx [hunk 1]
```diff
@@ -13,60 +13,93 @@ import {
...
+            setType("password")
             setShowPassword(false)
         }
     }
 
+    const handleLogin = () => {
+        login(email, password)
+        .then((data) => {
+            if (data.status === 200) {
+                navigate("/home");
```
Comment (Ekaterina-Bondareva) [inline src/components/Login.tsx:46]: Could you please move state cleanup here.

File: src/components/Login.tsx [hunk 2]
```diff
@@ -13,60 +13,93 @@ import {
...
+            }
+        })
+        .catch((error) => {
+            console.log(error);
+            if (error) {
```
Comment (Ekaterina-Bondareva) [inline src/components/Login.tsx:51]: Are you going to navigate user on page 404 for any error? How are you going to handle wrong credential case?

File: src/components/Register.tsx [hunk 3]
```diff
@@ -15,58 +15,90 @@ import {
...
+      setTypeConfirm("password")
       setShowPassword(false)
     }
   }
 
+  const handleRegister = () => {
+    register(name, email, password, confirmPassword) 
+    .then((data) => {
+      if (data.status === 201) {
+        navigate("/home");
```
Comment (Ekaterina-Bondareva) [inline src/components/Register.tsx:55]: Could you please move state cleanup here?

File: src/components/Register.tsx [hunk 4]
```diff
@@ -15,58 +15,90 @@ import {
...
+      }
+    })
+    .catch((error) => {
```
Comment (Ekaterina-Bondareva) [inline src/components/Register.tsx:58]: The same question about error handling.

**Review by Ekaterina-Bondareva — CHANGES_REQUESTED** [review]
Good job! PFA comments.

File: src/components/Login.tsx (hunk 2 above)
Comment (AnnaSolovykh) [reply src/components/Login.tsx:51]: Error handling for other issue will be the next ticket as it is said in the description.

File: src/components/Register.tsx (hunk 3 above)
Comment (AnnaSolovykh) [reply src/components/Register.tsx:55]: Thank you! Right, it makes sense!

**Review by AnnaPestova1 — APPROVED** [review]
I can register and login. everything works. Also received error message when didn't put credentials

**Review by Ekaterina-Bondareva — APPROVED** [review]

**Pull request merged**